packages/neuron-wallet/src/services/light-runner.ts

```diff
@@ -0,0 +1,148 @@
+import env from 'env'
+import path from 'path'
+import fs from 'fs'
+import { ChildProcess, spawn } from 'child_process'
+import logger from 'utils/logger'
+import { resetSyncTaskQueue } from 'block-sync-renderer'
+import SettingsService from 'services/settings'
+
+const { app } = env
+
+export enum NetworkType {
+  Light,
+  Full
+}
+
+abstract class NodeRunner {
+  protected constructor() {}
+  protected abstract networkType: NetworkType
+  protected runnerProcess: ChildProcess | undefined
+  protected static instance: NodeRunner | undefined
+  static getInstance(): NodeRunner {
+    throw new Error('should be called by instance')
+  }
+
+  protected abstract get binary(): string
+  abstract start(): Promise<void>
+  platform(): string {
+    switch (process.platform) {
+      case 'win32':
+        return 'win'
+      case 'linux':
+        return 'linux'
+      case 'darwin':
+        return 'mac'
+      default:
+        return ''
+    }
+  }
+
+  async stop() {
+    return new Promise<void>(resolve => {
+      resetSyncTaskQueue.push(false)
+      if (this.runnerProcess) {
+        logger.info('Runner:\tkilling node')
+        this.runnerProcess.once('close', () => resolve())
+        this.runnerProcess.kill('SIGKILL')
+        this.runnerProcess = undefined
+      } else {
+        resolve()
+      }
+    })
+  }
+}
+
+export class CKBLightRunner extends NodeRunner {
+  protected networkType: NetworkType = NetworkType.Full
+
+  static getInstance(): CKBLightRunner {
+    if (!CKBLightRunner.instance) {
+      CKBLightRunner.instance = new CKBLightRunner()
+    }
+    return CKBLightRunner.instance as CKBLightRunner
+  }
+
+  protected get binary() {
+    const appPath = app.isPackaged
+      ? path.join(path.dirname(app.getAppPath()), '..', './bin')
+      : path.join(__dirname, '../../bin')
+    const binary = app.isPackaged
+      ? path.resolve(appPath, './ckb-light-client')
+      : path.resolve(appPath, `./${this.platform()}`, './ckb-light-client')
+    return this.platform() === 'win' ? binary + '.exe' : binary
+  }
```

Comment:
There should be a similar logic to download `ckb binary`, maybe they can be encapsulated